crates/re_viewer/src/ui/top_panel.rs

```diff
@@ -261,46 +260,49 @@ fn panel_buttons_r2l(app: &App, app_blueprint: &AppBlueprint<'_>, ui: &mut egui:
     }
 
     // selection panel
-    if ui
-        .medium_icon_toggle_button(
-            &re_ui::icons::RIGHT_PANEL_TOGGLE,
-            &mut app_blueprint.selection_panel_state().is_expanded(),
-        )
-        .on_hover_text(format!(
-            "Toggle Selection View{}",
-            UICommand::ToggleSelectionPanel.format_shortcut_tooltip_suffix(ui.ctx())
-        ))
-        .clicked()
+    if !app_blueprint.selection_panel_overridden()
```

Comment:
Not for this PR since the logic probably adds too much complexity, but I think we may want to reconsider this at some point and have this be something like: `!app_blueprint.selection_panel_hidden()`

It would be nice if the state was determine by:
 - If hidden, never show it -- no UI controls.
 - If user has interacted with the controls, use that value.
 - If overridden from JS, apply that value.
 - If the blueprint has a value, apply that value.